src/internal/components/cartesian-chart/bottom-labels.tsx

```diff
@@ -101,19 +113,41 @@ function BottomLabels({
               aria-label={lines.join('\n')}
             >
               <line className={styles.ticks__line} x1={0} x2={0} y1={0} y2={TICK_LENGTH} aria-hidden="true" />
-              {lines.map((line, lineIndex) => (
-                <text
-                  className={styles.ticks__text}
-                  key={lineIndex}
-                  x={0}
-                  y={TICK_LENGTH + TICK_MARGIN + lineIndex * TICK_LINE_HEIGHT}
+              {isCategorical ? (
```

Comment:
new property will be added, so that users can control behaviour themselves:

`isCategorical && categoricalLabelOverflowType === 'ellipsis' ? .. : ..`